docs/rfc/rfc-6.adoc

```diff
@@ -0,0 +1,228 @@
+functionality built-in and enabled in the configuration file.
+
+We will implement a simple reimbursement contract for maintainers proxying calls
+to Bridge functions. Only approved maintainers should be able to use the proxy
+contract. Maintainers are expected to follow round-robin distribution of work
+but the order is not going to be enforced on-chain. DAO/council will have the
+functionality of removing maintainers not performing their duties.
+DAO/council may decide to distribute additional T rewards for maintainers. This
+functionality should be supported by the Maintainer contract.
+
+Eventually, the role of maintainers can be delegated to another network such as
+Gelato.
+
+
+```
+┌────────────────────────────────────────┐   calls  ┌─────────────────────────┐
+|               Maintainer               |─────────▶|          Bridge         |
+└────────────────────────────────────────┘          └─────────────────────────┘  
+| submitSweeProof() onlyMaintainer       |          | submitSweepProof()      | 
+| submitRedemptionProof() onlyMaintainer |          | submitRedemptionProof() | 
+| startDkg() onlyMaintainer              |          | startDkg()              | 
+└────────────────────────────────────────┘          └─────────────────────────┘
+```
+
+===== Submit sweep proof
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+===== Submit redemption proof
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+===== Start DKG
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers. 
+
+===== Report DKG timeout
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+==== Third party incentives
+
+
+All transactions in this section must be supported by the off-chain client. We
+are not concerned about front-running. Front-running may happen and it is even
+desirable to some extent because these transactions protect the health of the
+network.
+
+[[challenge-dkg-result]]
+===== Challenge DKG result
+
+No ETH cost reimbursed. Called exceptionally, ideally never. Incentivized by T
+from slashed stakes.
+
+===== Notify redemption timeout
+
+
+No ETH cost reimbursed. Called exceptionally, ideally never. Incentivized by T
+from slashed stakes.
+
+[[report-fraud]]
+===== Report fraud
+
+No ETH cost reimbursed. Called exceptionally, ideally never. Incentivized by T
+from slashed stakes.
+
+=== ETH pool
+
+DAO needs to fund ETH pool that will be used for reimbursements. The pool should
+probably be a separate contract. The pool needs to protect against malicious
+miner-operators by placing a governable gas price ceiling. It should be possible
+to withdraw unspent ETH in case we decide to replace Maintainer incentives with
+something else (for example Gelato) or decide to add more functions there
+(for example, the SPV relay updates). 
```

Comment:
Maybe it would be worth mentioning that ETH Pool should be upgradeable (it could be connected with the possibility to withdraw unspent ETH, e.g. by transferring it to a new pool)

Reply:
> Maybe it would be worth mentioning that ETH Pool should be upgradeable (it could be connected with the possibility to withdraw unspent ETH, e.g. by transferring it to a new pool)

It does not have to be upgradeable but yes, it needs to have a function allowing the DAO to withdraw unspent ETH. I was thinking this is obvious but if not, I can make it clear in the RFC.